### PR TITLE
Migrating events from SharePoint to TinaCMS

### LINKED ISSUE #2694: 🚶 Events - migrate SharePoint list data to Tina
Currently the Events are sourced from a SharePoint list - this isn't ideal since:

- we need to query an external source
- it makes it harder to create a new event
  first add an image to the website, then add data to a list
  if you do it too quickly, it can cause tests to fail on deployment due to broken images
- this will help perf stats as next will be able to optimise them

```[tasklist]
### Tasks
- [x] Create `event` collection
- [x] Write script to export SharePoint event list and presenter data
- [x] Standardise event thumbnail location (in schema)
- [x] Reference TinaCMS event data for events and Live UG
- [x] Remove unused environment variables from build
- [x] Update marketing standard (give Camilla a video)
```

### PR DESCRIPTION
**Pull request by andrewwaltosssw**
- Migrated all past and future events (804) to TinaCMS as json data.
- Created new Collection to store events
- Updated existing references to SharePoint event data to query TinaCMS instead
- Removed env vars for SharePoint
- Enabled folder creation in Tina to allow events to be created in future years
---
- Affected routes: `/`, `/live`, `events`, `netug/*`

- Fixed #2694 

- [x] Include done video or screenshots
image

### CODE REVIEW

File: app/live-steam-banner/liveStreamWidget.tsx
Comment (amankumarrr) [inline app/live-steam-banner/liveStreamWidget.tsx]: Maybe we should change this to Rich-text to make it easy for the end user to edit things.

File: components/filter/events.tsx
Comment (amankumarrr) [inline components/filter/events.tsx]: maybe we should import it something like @/hook/*.

File: pages/events/index.tsx
Comment (amankumarrr) [inline pages/events/index.tsx]: we should look into HydrationBoundary if it is still relevant

**Review by amankumarrr — CHANGES_REQUESTED** [review]
Please see the comments

File: app/live-steam-banner/liveStreamWidget.tsx
Comment (awaltos) [reply app/live-steam-banner/liveStreamWidget.tsx]: Done - updated to rich-text field in collection and references

File: components/filter/events.tsx
Comment (bradystroud) [inline components/filter/events.tsx]: Assuming this is the presenter name, change to presenterName
```suggestion
  presenterName?: string;
```

File: graphql.config.js
Comment (bradystroud) [inline graphql.config.js:1]: not sure what caused this change

File: app/live-steam-banner/liveStreamWidget.tsx
Comment (bradystroud) [inline app/live-steam-banner/liveStreamWidget.tsx]: thats lots of presenters... Can we simplify that data structure?

if not maybe at the top you can do

```ts
const presenter = presenter.presenter;
```

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: components/filter/events.tsx
Comment (awaltos) [reply components/filter/events.tsx]: done - updated to use aliases

File: graphql.config.js
Comment (awaltos) [reply graphql.config.js:1]: We moved tina files from `.tina` to `tina` on https://github.com/SSWConsulting/SSW.Website/issues/2422
I'm updating the references to gql files here

File: app/live-steam-banner/liveStreamWidget.tsx
Comment (awaltos) [reply app/live-steam-banner/liveStreamWidget.tsx]: Done - did the second thing. Data structure + gql result structure is causing this repetition

File: components/filter/events.tsx
Comment (awaltos) [reply components/filter/events.tsx]: Reverted these changes - Build was failing probably due to the issues you raised in this issue: https://github.com/tinacms/tinacms/issues/4488

File: components/filter/events.tsx
Comment (awaltos) [reply components/filter/events.tsx]: Done

File: pages/events/index.tsx
Comment (awaltos) [reply pages/events/index.tsx]: added hydration boundary and prefetching of query in static props as previously implemented

**Comment by andrewwaltosssw** [comment]
Points after discussion with @camillars and @sethdaily to demo the new flow of adding evetns via Tina, the following points were brought up. Please see their notes for actions. These points are nice-to-haves and can be actioned at a later stage.

- [ ] Can we add event start date in columns on events view? - **Note:** Likely this would be a TinaCMS change - if at all possible
image
**Figure: Events list view where it would be helpful to see the event date**
- [ ] Live Stream: this is always ssw.com.au/live and desc is always Live Stream. Maybe change to boolean and make description text say "Will this event have a live stream?". **Note:** We don't render this property anywhere on v3 website. Either use the property somewhere or remove it from the collection if not needed.
- [x] Calendar Type: Remove firebootcamp, uts, webinars, and solid Q. **Note:** Done
- [x] City: Add free text option for Other **Note:** Done
- [x] Enabled: Add description "Show this event on ssw.com.au **Note:** Done
- [ ] Category: Seth feels it is not always easy to select the correct option here. Update them by making them more generic
- [x] Description: Add a description of "This shows on ssw.com.au/events **Note:** Done
- [x] Internal note: Cam + Seth don't use this. Add a description of "Note - this does not render anywhere" move to bottom **Note:** Done
- [x] Delayed livestream start: Camilla is repeating work when creating some events. Implement the following rule: if (Live stream event && Category === "User Groups") then enable Delayed LiveStream and set value to 30. **Note:** this would require a new boolean for livestreamEvent - see second checkpoint here

cc: @Calinator444 @bradystroud @wicksipedia

File: content/events-calendar/2006/Amazon-web-services.json
Comment (bradystroud) [inline content/events-calendar/2006/Amazon-web-services.json]: (this applies to all migrated events)
Confused why this is a URL... I expected some alt text.

I'd remove the URL.

File: content/events-calendar/2006/Amazon-web-services.json
Comment (bradystroud) [inline content/events-calendar/2006/Amazon-web-services.json]: (this applies to all migrated events)
All links should be relative

File: content/events-calendar/2024/Unlocking-the-Full-Potential-The-Harmonious-Dance-of-EF-Core-and-SQL-Server.json
Comment (bradystroud) [inline content/events-calendar/2024/Unlocking-the-Full-Potential-The-Harmonious-Dance-of-EF-Core-and-SQL-Server.json:7]: Confused why we have these two fields when we also have jernej-kavka.mdx

~I would expect this to be a reference type to the presenter file instead~

Line 39 has the presenters list... I think we can remove these fields and use that instead

File: tina/collections/presenter.tsx
Comment (bradystroud) [inline tina/collections/presenter.tsx:1]: This data structure seems wrong. why is the Name and URL nested?

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: content/events-calendar/2006/Amazon-web-services.json
Comment (awaltos) [reply content/events-calendar/2006/Amazon-web-services.json]: This is the data as migrated from sharepoint. Matt mentioned that does not want to lose any data. 

Perhaps we could just populate this with the event title if a url was used

File: content/events-calendar/2024/Unlocking-the-Full-Potential-The-Harmonious-Dance-of-EF-Core-and-SQL-Server.json
Comment (awaltos) [reply content/events-calendar/2024/Unlocking-the-Full-Potential-The-Harmonious-Dance-of-EF-Core-and-SQL-Server.json:7]: There are cases where we have presenterName but nothing in presentersList. 
This might suit the case when marketing does not want to add a new presenter to presenters collection (i.e. a one-off)

File: tina/collections/presenter.tsx
Comment (awaltos) [reply tina/collections/presenter.tsx:1]: This is the existing structure. Could possibly be flattened but would impact other areas and need additional refactoring.

**Review by amankumarrr — APPROVED** [review]
Approving this PR to unblock it as I'm going on client work again.

File: content/events-calendar/2006/Amazon-web-services.json
Comment (bradystroud) [reply content/events-calendar/2006/Amazon-web-services.json]: Fixed by @Calinator444

**Review by bradystroud — CHANGES_REQUESTED** [review]
Hey @Calinator444 As per our conversation, we cannot merge this until the marketing team is aware of how to use it.

- [ ] make a video and supporting docs
- [x] Get a 'test pass' from someone at marketing

**Review by bradystroud — APPROVED** [review]
LGTM - performance issues seem to be fixed.

**Review by wicksipedia — APPROVED** [review]
`.github/` changes lgtm

**Pull request merged**
